### DISCUSSION TITLE
Node.js compilation for Windows x86 target architecture fails with code -1073741819

### DISCUSSION THREAD

**Discussion by vladimir-andreevich** [nodejs/help#4718] [Q&A]
I have encountered troubles while building Node.js on 64-bit Microsoft Windows 19044.2728. The x64 executable and x64 MSI installer are built successfully. However, when I try to build Node.js for x86 target architecture, the system after some time of compilation outputs the following error:
`C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\MSBuild\Microsoft\VC\v160\Microsoft.CppCommon.targets(241,5): error MSB8066: Custom build for 'out\Release\\mkcodecache.exe;out\Release\\node_mksnapshot.exe' exited with code -1073741819. [node-14.19.3\node.vcxproj]`
I performed a search and did not manage to find much information about this specific problem. Could you help to resolve this issue?

**Comment by bnoordhuis** [comment]
-1073741819 is 0xC0000005 is Access Violation, i.e., most likely a null or bad pointer dereference. Try building v14.21.3 instead (or the latest v16.x or v18.x); you may be hitting a bug that's already been fixed.

**Reply by vladimir-andreevich** [reply]
I tried to build v14.21.3, and I faced the same problem. The x64 executable and x64 MSI installer are built successfully; the build for the x86 architecture aborts with the same error. I did not make any changes to the Node.js source code.

**Reply by bnoordhuis** [reply]
Our CI still seems to build it fine so it's likely something on your end, although I can't speculate on what exactly.

v14.x is going end-of-life at the end of the month so you may want to factor that into your decision on whether to sink more time in it.

**Reply by vladimir-andreevich** [reply]
Indeed, something happens on my end, and I try to figure out what happens. The reason is that the system does not provide any additional information. Is there a way to get details of the error and to discover where exactly the error occurs with debug tools?

**Reply by bnoordhuis** [reply]
If you can grab the stack trace with a C++ debugger and post it here, I can probably point you in the right direction.

**Reply by vladimir-andreevich** [reply]
Unfortunately, LLDB debugger, which I use, refuses to attach to `vcbuild.bat`: it complains that it cannot find any 'host' platform architecture in the file. Currently, I am unable to grab the stack trace. I had to find the problem myself. I discovered that the error happens at the line `msbuild node.sln %msbcpu% /t:%target% /p:Configuration=%config% /p:Platform=%msbplatform% /clp:NoItemAndPropertyList;Verbosity=minimal /nologo %extra_msbuild_args%`. In my case, the command during the build is `msbuild node.sln /m:2 /t:"Build" /p:Configuration=Release /p:Platform=Win32 /clp:NoItemAndPropertyList;Verbosity=minimal /nologo`. It is now clear why the system does not provide much information about the error: the output verbosity is set to minimal. I changed the level of verbosity to `detailed`, and the system provided a huge log. The end of this log is attached to this message because the original log is 43 MB. I hope this information might be helpful.

[vcbuild_log.txt](https://github.com/nodejs/node/files/11269365/vcbuild_log.txt)

**Reply by bnoordhuis** [reply]
Look up the node_mksnapshot invocation in the sln file and run that directly under your debugger. You can also piece it together from the node.gyp file in the top-level directory.

**Reply by vladimir-andreevich** [reply]
It took me a while to make it. If launched separately with the command `msbuild node_mksnapshot.vcxproj /m:2 /t:"Build" /p:Configuration=Release /p:Platform=Win32`, node_mksnapshot builds successfully for Windows x86 target architecture. The error occurs only during the whole application build. This is what LLDB shows me when I attach it to the MSBuild process while building Node.js with the command `vcbuild.bat release x86 msi`:

[lldb_msbuild.txt](https://github.com/nodejs/node/files/11351473/lldb_msbuild.txt)

I hope this information is helpful.

**Reply by bnoordhuis** [reply]
It's devoid of details, I'm afraid. Not through any fault of yours, the stack frames simply aren't there.

My hunch is that it's crashing in the machine code that V8 emits because that uses a calling convention that C++ debuggers don't understand. Such bugs are unfortunately very hard to debug.

**Reply by vladimir-andreevich** [reply]
I believe this case needs further investigation. Do you need any additional information while I can still reproduce the issue?

**Reply by bnoordhuis** [reply]
Support for x86 Windows was demoted to experimental status a while ago, meaning: we accept fixes but don't expect anyone to investigate.
